packages/sync-rules/src/TsSchemaGenerator.ts

```diff
@@ -3,26 +3,53 @@ import { SchemaGenerator } from './SchemaGenerator.js';
 import { SqlSyncRules } from './SqlSyncRules.js';
 import { SourceSchema } from './types.js';
 
+export interface TsSchemaGeneratorOptions {
+  language?: 'ts' | 'js';
+  imports?: 'web' | 'react-native' | 'auto';
+}
 export class TsSchemaGenerator extends SchemaGenerator {
   readonly key = 'ts';
   readonly label = 'TypeScript';
   readonly mediaType = 'application/typescript';
   readonly fileName = 'schema.ts';
 
+  constructor(public readonly options: TsSchemaGeneratorOptions = {}) {
+    super();
+  }
+
   generate(source: SqlSyncRules, schema: SourceSchema): string {
     const tables = super.getAllTables(source, schema);
 
-    return `import { column, Schema, TableV2 } from '@powersync/web';
-// OR: import { column, Schema, TableV2 } from '@powersync/react-native';
+    return `${this.generateImports()}
 
 ${tables.map((table) => this.generateTable(table.name, table.columns)).join('\n\n')}
 
 export const AppSchema = new Schema({
   ${tables.map((table) => table.name).join(',\n  ')}
 });
 
-export type Database = (typeof AppSchema)['types'];
-`;
+${this.generateTypeExports()}`;
+  }
+
+  private generateTypeExports() {
+    const lang = this.options.language ?? 'ts';
+    if (lang == 'ts') {
+      return `export type Database = (typeof AppSchema)['types'];\n`;
+    } else {
+      return ``;
+    }
+  }
+
+  private generateImports() {
+    const importStyle = this.options.imports ?? 'auto';
+    if (importStyle == 'web') {
```

Comment:
If we use enums we can also add some nice type safety.

I think a map here would be cleaner e.g.:
```
enum ImportStyle {
  Web = 'web',
  ReactNative = 'react-native',
  Auto = 'auto'
}
...

styleMap: Record<ImportStyle, string> = {
 [ImportStyle.Web]: `import { column, Schema, TableV2 } from '@powersync/web';`,
 [ImportStyle.ReactNative]: `import { column, Schema, TableV2 } from '@powersync/react-native';`
 [ImportStyle.Auto]: `import { column, Schema, TableV2 } from '@powersync/web';
// OR: import { column, Schema, TableV2 } from '@powersync/react-native';`
}

return styleMap[importStyle]
}
```

Reply:
Note that string literal union types gives just as much type safety as enums, but I switched to proper enums here.

I'm still keeping the same style of if/else in the function - it's more flexible if we need to customize it more in the future.